kubespawner/spawner.py

```diff
@@ -583,15 +583,19 @@ def _hub_connect_port_default(self):
     )
 
     def _expand_user_properties(self, template):
-        # Make sure username matches the restrictions for DNS labels
+        # Make sure username and servername match the restrictions for DNS labels
         safe_chars = set(string.ascii_lowercase + string.digits)
+        # Set servername based on whether named-server initialised
+        temp_name = getattr(self, 'name', '')
+        if temp_name:
+            server_name = '-' + temp_name
+        else:
+            server_name = ''
+
         legacy_escaped_username = ''.join([s if s in safe_chars else '-' for s in self.user.name.lower()])
         safe_username = escapism.escape(self.user.name, safe=safe_chars, escape_char='-').lower()
-        return template.format(
-            userid=self.user.id,
-            username=safe_username,
-            legacy_escape_username=legacy_escaped_username
-        )
+        d = {'username': safe_username, 'servername': server_name}
```

Comment:
Instead of rewriting this dict, you can just add `servername=server_name` to the original dict here. As it is now, a few keys have been lost and it's hardcoded `self.pod_name_template` instead of `template`

Reply:
I tried the other approach every which way and couldn't get it to work.